server.sml

```diff
@@ -121,12 +121,16 @@ fun finish id =
     val sha = get_head_sha (read_bare_snapshot inp)
     val status = read_status inp
     val inp = (TextIO.closeIn inp; TextIO.openIn new)
-    val typ = read_job_type inp
+    val pr_info = read_job_pr inp
+    val branch =
+      case read_job_pr inp of
+        NONE => "master"
+      | SOME (_, branch) => Substring.string branch
```

Comment:
I think `branch` might contain parentheses, at least  in the files you sent me the format was `#<PR> (<branch>)`